calling-apis/post-requests-with-axios.md

```diff
@@ -0,0 +1,221 @@
+1. Required: `name`, `continent`, `category`, `weeks`, and `cost`. Optional: `about`
+
+</details>
+
+## Making a `POST` request
+
+The syntax for making a POST request using `axios` is very similar to making a GET request, but we will use the `post` function instead.
+
+### An Aside: Working With Ambiguous Documentation
+
+What happens if Matteo [finds that the official `POST` request documentation is ambiguous](https://axios-http.com/docs/post_example), and he can't find more satisfactory resources?
+
+For example, Matteo could go to the POST request page and see this code, with zero explanation:
+
+<!-- prettier-ignore-start -->
+```js
+axios.post('/user', {
+    firstName: 'Fred',
+    lastName: 'Flintstone'
+  })
+  .then(function (response) {
+      console.log(response);
+  })
+  .catch(function (error) {
+      console.log(error);
+  });
+```
+<!-- prettier-ignore-end -->
+
+The best thing for Matteo to do in this situation is:
+
+1. Make hypotheses, using context clues and what he knows about APIs
+1. Test them
+
+From this example code, we can see that the second argument for `axios.post` is this object literal:
+
+<!-- prettier-ignore-start -->
+```js
+{
+    firstName: 'Fred',
+    lastName: 'Flintstone'
+}
+```
+<!-- prettier-ignore-end -->
+
+Matteo sees that there are two keys, `firstName` and `lastName`. From his knowledge of APIs, these keys don't look like they're describing parts of an HTTP request like HTTP headers or params. Additionally, this example request is being made to `/user`.
+
+Matteo hypothesizes that this object literal holds the HTTP request body data, and he can test his hypothesis against his API.
+
+### Sending a Request to the Ada Trip Reserving Service™ Trip API™™™
+
+Matteo is ready to try to make a `POST` request to the back-end service. He [determines from the API documentation](https://github.com/AdaGold/trip-api) that the request he wants to make is:
+
+- `POST` request to `https://trektravel.herokuapp.com/trips`
+- Request Body:
+  - `name`
+  - `continent`
+  - `about`
+  - `category`
+  - `weeks`
+  - `cost`
+
+The `post` function takes in two parameters: URL, and an optional object. This optional object represents the HTTP request body data that we want to send.
+
+Matteo could write this code:
+
+```js
+const axios = require("axios");
+
+const tripData = {
+  name: "Matteo's Chill Trip to Chicago",
+  continent: "North America",
+  about: "A tour around good architecture and hot dogs.",
+  category: "casual",
+  weeks: 1,
+  cost: 180,
+};
+
+axios
+  .post("https://trektravel.herokuapp.com/trips", tripData)
+  .then((response) => {
+    console.log("response:", response);
+    console.log("response data:", response.data);
+  })
+  .catch((error) => {
+    console.log("error:", error);
+    console.log("error response:", error.response);
+  })
+  .finally(() => {
+    console.log("finally done!");
+  });
+```
+
+With this code, Matteo can begin to make more `POST` requests in JavaScript!
+
```

Comment:
if the reader is following along and runs the code, they'll get an error. add a callout to explain why
```suggestion

### !callout-warning

## Trip Names Must Be Unique

If we try running Matteo's code, we're likely to get an error! The Trip API requires the trip package names to be unique, and Matteo has already used the name in the example. But we can supply a new name to try running the code ourselves!

### !end-callout

```

Reply:
I actually didn't experience this... Hm, weird

Reply:
I just double-checked. If I post a trip with the same name, I get a 400. And I remember checking in the model and seeing there's a uniqueness constraint on the name.